process/work_tracking/milestones.md

```diff
@@ -0,0 +1,34 @@
+
+If an issue tracks a bug fix in a released stable version of the product, then it should have a `bug` label assigned and the team should come to a conclusion for how far back of a version (the oldest support version of the product) the bug fix should target. This determines the patch milestone to assign to the issue. For example, consider a scenario within the `leap` repo where where versions 3.2 and 4.0 of Leap are supported while the `main` branch is tracking work for an upcoming 5.0 release. If a bug is discovered that impacts code as old as version 3.2, the issue may be assigned a milestone `Leap 3.2.Z` (where `Z` is replaced by the next available patch version number within the 3.2 series that has yet to be released). This also signals to the developer that they must first target to bug fix to the `release/3.2` branch, then bring the fix forward to the `release/4.0` branch, and then finally bring the fix forward to the `main` branch. If another bug is discovered that impacts version 4.0 but not any older versions, then the issue may be assigned a milestone `Leap 4.0.Z` (where again `Z` is set appropriately to the next available patch version number in the 4.0 series).
+
+If an issue tracks a bug fix in a release candidate or it includes other refinements to the release candidate prior to the stable release, e.g. documentation updates, then it should be assigned the appropriate patch milestone. For example, consider work on top of a recently released Leap 5.0.0-rc1. If a bug was discovered in Leap 5.0.0-rc1 that requires a new release candidate to test, then the issue tracking the bug would be assigned the `Leap 5.0.0-rc2` patch milestone. There may also be some issues that are not considered bugs but must be completed prior to being able to release a stable version of 5.0.0, e.g. documentation updates; those issues would be assigned the `Leap 5.0.0` patch milestone.
+
+If an issue is not a bug fix or changes for an existing release candidate, e.g. if it is an enhancement or new feature, then it should be assigned the appropriate feature milestone. For example, consider a scenario within the `leap` repo where the `main` branch is tracking work for an upcoming 5.0 release of Leap. A new issue tracking a new feature for 5.0 would likely be assigned the `Leap 5.0.0-rc1` feature milestone, unless there is an earlier and more granular feature milestone for it that is more appropriate.
+
+## Assigning milestones to PRs
+
+Pull requests typically should not have milestones attached to them. The only exception is for the PRs that forward bug fixes from a release branch forward to another release branch. In that case, the PR forwarding to a release branch should be assigned a patch milestone corresponding to the version of the target release branch.
```

Comment:
The only uncertainty I have is whether PRs _without a linked issue_ also play by the same rules here: no milestone?

Reply:
I'm not sure. I left it ambiguous for now. I guess that is something we can figure out later as we further refine this process.